### ISSUE TITLE
Wrong inference maybe due to inference cache

### ISSUE THREAD

**Issue by hippo91** [PyCQA/astroid#669]
This bug has been found during the setup of the `numpy` brains. 
Fixing it could improve our inference system for `numpy` functions.

The directory described hereafter reflects a similar structure inside the `numpy` sources.
### Steps to reproduce
1. Create a directory named `bug_inf` with the following structure:
```
$ tree bug_inf
bug_inf
├── core
│   ├── __init__.py
│   └── shape_base.py
├── __init__.py
└── lib
    ├── __init__.py
    └── shape_base.py
```
2. `bug_inf/core/__init__.py` content is:
```
$ cat bug_inf/core/__init__.py 
from .shape_base import *
```
3. `bug_inf/core/shape_base.py` content is:
```
$ cat bug_inf/core/shape_base.py 
def vstack():
    return 42
```
4. `bug_inf/__init__.py` content is:
```
$ cat bug_inf/__init__.py 
from .core import *
from .lib import *
```
5. `bug_inf/lib/__init__.py` content is:
```
$ cat bug_inf/lib/__init__.py 
from .shape_base import *
```
6. `bug_inf/lib/shape_base.py` content is:
```
$ cat bug_inf/lib/shape_base.py 
from bug_inf.core import vstack
```
7. Create a file `bug_inf.py` which content is:
```
from astroid import extract_node

SRC = """
from bug_inf import vstack
vstack()
"""

NODE = extract_node(SRC)
print(NODE.inferred())
```
8. Run `python3 bug_inf.py`

### Current behavior
```
 $ python3 bug_inf.py 
[<Const.int l.2 at 0x7f08471338d0>, Uninferable]
```

### Expected behavior
```
 $ python3 bug_inf.py 
[<Const.int l.2 at 0x7f08471338d0>]
```

### Astroid version
```
python3 -c "from astroid import __pkginfo__; print(__pkginfo__.version)"
2.3.0
```

**Comment by hippo91** [comment]
Here is attached an archive with the files described in the issue.
[bug_inf.tar.gz](https://github.com/PyCQA/astroid/files/4009900/bug_inf.tar.gz)

**Comment by hippo91** [comment]
Fixed thanks to PR #884
